### PR TITLE
Removed all local keystores, added configurable trust dir

### PR DESCRIPTION
**Pull request by diogomonica**
Started the work to support a better cli.
- Removed local truststores
- Added configurable base directory
- Added configurable remote server
- Fixed key listing/removal

Signed-off-by: Diogo Monica <diogo@docker.com>

### CODE REVIEW

File: cmd/notary/keys.go [hunk 1]
```diff
@@ -4,104 +4,63 @@ import (
...
-		err = certificateStore.RemoveCert(cert)
-		if err != nil {
-			fatalf("failed to remove certificate from KeyStore")
-		}
-		return
+	keyID := args[0]
+	if len(keyID) != 64 {
+		fatalf("please enter a valid root key ID")
 	}
+	parseConfig()
```
Comment (aaronlehmann) [inline cmd/notary/keys.go]: Would it make sense to call parseConfig() in main so we don't have to call it in every command function?

File: cmd/notary/keys.go [hunk 2]
```diff
@@ -4,104 +4,63 @@ import (
...
+	if err != nil {
+		fatalf("failed to create a new truststore manager with directory: %s", trustDir)
 	}
 
 	// List all the keys about to be removed
-	fmt.Println("Are you sure you want to remove the following keys? (yes/no)", gunOrID)
-	for _, k := range keyList {
-		printKey(k)
-	}
+	fmt.Printf("Are you sure you want to remove the following key? (yes/no)\n%s\n", keyID)
```
Comment (aaronlehmann) [inline cmd/notary/keys.go]: Move the (yes/no) to after the key ID?

File: cmd/notary/main.go [hunk 3]
```diff
@@ -1,96 +1,71 @@
...
-	"github.com/docker/notary/trustmanager"
+	"github.com/docker/notary/pkg/passphrase"
 )
 
 const configFileName string = "config"
-const configPath string = ".docker/trust/"
-const trustDir string = "trusted_certificates/"
-const privDir string = "private/"
-const rootKeysDir string = "root_keys/"
+const defaultTrustDir string = ".notary/"
```
Comment (aaronlehmann) [inline cmd/notary/main.go]: I realize that we don't want to tie this tool to docker, but this means that you need to override the default trust dir to use it with docker, right? Isn't that awkward UX?

File: cmd/notary/main.go [hunk 4]
```diff
@@ -1,96 +1,71 @@
...
+const defaultServerURL = "https://notary-server:4443"
```
Comment (aaronlehmann) [inline cmd/notary/main.go]: This is kind of weird too. notary-server won't exist unless someone adds it to a host file. Maybe it's better to have no default, and require the user to specify the server on the command line or in the config file. The error message would be more helpful than a message about not being able to resolve notary-server.

File: cmd/notary/main.go (hunk 3 above)
Comment (diogomonica) [inline cmd/notary/main.go]: Any proposals on what to do?

People could add a configuration file to config.json under ~./notary to always point it to ~/.docker/trust

File: cmd/notary/keys.go (hunk 1 above)
Comment (diogomonica) [inline cmd/notary/keys.go]: parseConfig has to be run after `Command.Execute` from viper, or else the command line flags (verbose, basedir, etc) won't be available.

Not sure if there is an alternative to calling it within every command context.

File: cmd/notary/keys.go (hunk 2 above)
Comment (diogomonica) [inline cmd/notary/keys.go]: done

File: cmd/notary/main.go (hunk 3 above)
Comment (diogomonica) [inline cmd/notary/main.go]: Not sure if having notary point at ~/.docker/trust by default is a good idea. People might screw things up by accident.

File: cmd/notary/main.go (hunk 4 above)
Comment (diogomonica) [inline cmd/notary/main.go]: We're thinking about adding a default server at thenotary.io that people can use as the default trust server. Unfortunately setting that up (and how we would enforce GUN ownership), is currently an unsolved issue.

**Comment by aaronlehmann** [comment]
LGTM, we'll address the https://notary-server thing later.

**Pull request merged**